orttraining/tools/scripts/opset12_model_transform.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+#
+# This converter is an internal util to upgrade existing bert/gpt-2 models, 
+# which were previously transformed/optimized from orginal model, to Opset 12 
+# version as well as replacing deprecated node, i.e., TrainableDropout with 
+# the "Dropout" node matching the Opset 12 Spec. Typically, a model to 
+# be run by this scripts would have "_optimized" substring in its model name, 
+# and the graph should have one or more "TrainableDropout" nodes in its graph. 
+# Example usage: 
+#   python opset12_model_transform.py bert-base-uncased_L_12_H_768_A_12_V_30528_S_512_Dp_0.1_optimized_layer_norm.onnx
+# Output: 
+#   bert-base-uncased_L_12_H_768_A_12_V_30528_S_512_Dp_0.1_optimized_layer_norm_opset12.onnx
+
+import sys
+import onnx
+from onnx import helper, shape_inference
+from onnx import TensorProto
+import numpy as np
+from onnx import numpy_helper
+
+if len(sys.argv) < 2:
+    print("Please give model path...")
+    exit(1)
+
+input_model_name = sys.argv[1]
+output_model_name = input_model_name[:-5] + '_opset12.onnx'
+
+model = onnx.load(input_model_name)
+
+def find_input_node(model, arg):
+    result = []
+    for node in model.graph.node:
+        for output in node.output:
+            if output == arg:
+                result.append(node)
+    return result[0] if len(result)== 1 else None
+
+def find_output_node(model, arg):
+    result = []
+    for node in model.graph.node:
+        for input in node.input:
+            if input == arg:
+                result.append(node)
+    return result[0] if len(result) == 1 else None
+
+def find_input(model, arg):
+    for initializer in model.graph.initializer:
+        if initializer.name == arg:
+            return initializer
+    return None
+
+def get_node_index(model, node):
+    i = 0
```

Comment:
>i = 0 [](start = 4, length = 5)

fyi, the enumerate function might be useful here. it provides an index as well.
[https://docs.python.org/3/library/functions.html#enumerate](https://docs.python.org/3/library/functions.html#enumerate)

Reply:
will do this in my next PR